### PR TITLE
sync: Fix SHADER_READ expansion

### PR DESCRIPTION
**Pull request by artem-lunarg**
Closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/7349

image

### CODE REVIEW

**Comment by artem-lunarg** [comment]
Additionally disabled `QSTransitionHazardsPreviousBatch_TimelineSemaphore` which should be fine since we have `QSTransitionHazardsPreviousBatch_BinarySemaphore` which does the same. There is a chance timeline version causes problems on android CI but not 100% sure.

File: layers/sync/sync_utils.cpp [hunk 1]
```diff
@@ -99,8 +99,8 @@ VkPipelineStageFlags2KHR ExpandPipelineStages(VkPipelineStageFlags2KHR stage_mas
 }
 
 static const auto kShaderReadExpandBits =
```
Comment (spencer-lunarg) [inline layers/sync/sync_utils.cpp:101]: we have this defined in the XML, for now I am happy with this change, but it is also vulnerable if a new access is added and not updated here... could we move this logic to be generated from the XML

```xml
<syncaccess name="VK_ACCESS_2_SHADER_READ_BIT" alias="VK_ACCESS_SHADER_READ_BIT">
      <syncequivalent access="VK_ACCESS_2_SHADER_SAMPLED_READ_BIT,VK_ACCESS_2_SHADER_STORAGE_READ_BIT,VK_ACCESS_2_SHADER_BINDING_TABLE_READ_BIT_KHR"/>
</syncaccess>
```

**Review by spencer-lunarg — APPROVED** [review]

File: layers/sync/sync_utils.cpp (hunk 1 above)
Comment (artem-lunarg) [reply layers/sync/sync_utils.cpp:101]: good point

Comment (artem-lunarg) [reply layers/sync/sync_utils.cpp:101]: will make a separate PR for this

**Pull request merged**